### ISSUE TITLE
#Self-XSS# Script able to proceed in Variable “UserName”

### ISSUE THREAD

**Issue by kotobukki** [codex-studio/nav#11]
/js/userdata.js
like
`function cgNameCN() {
    userName = prompt("我应该称呼您什么?", "");
    if (userName != "" && userName != null) {
        setCookie("codexNavUsername", userName, 365);
        Materialize.toast('昵称成功更改为了 ' + userName + "!", 4000);
        document.getElementById("userID").innerHTML = "由CodEX软件工作室为" + userName + "量身定做!";
    }
    else {
        setCookie("codexNavUsername", "", 365);
        Materialize.toast('昵称删除成功!', 4000);
        document.getElementById("userID").innerHTML = "";
        $("#cgName").hide();
        $("#addName").show();
    }
}`
UserName is not flitered
# Sugestion：
1. cookie to Httponly
2.`<script src="https://raw.github.com/leizongmin/js-xss/master/build/xss.js"></script>
<script>
// 使用函数名 filterXSS，用法一样
var html = filterXSS('<script>alert("xss");</scr' + 'ipt>');
alert(html);
</script>`
3.use document.write

**Comment by mj66** [comment]
Please suggest pull request so that I can see the details of this suggestion. BTW, 2.x development is ending 7/20. If you want to get it into 2.x you have to submit your pull by 7/18 EDT. Otherwise it's 3.0 which will start 7/25.

**Comment by casprussian** [comment]
The client side is not secure.
But what does the XSS matter? All the code is on client side, yes?

**Comment by kotobukki** [comment]
yep but it can be used as phishing like i can said special effect after entering <script>xxxx<script> then get user information.Moreover after more complicate system in use, it may be more harmful.

**Comment by casprussian** [comment]
But that code will be limited to one machine? ITs not like the code that is injected will be written back to the server copy.

**Comment by kotobukki** [comment]
yep.Nothing leave on server.
https://en.wikipedia.org/wiki/Self-XSS

**Comment by mj66** [comment]
@kotobukki @casprussian 
CodEX Navigation is not designed to be secure. We do not leave any user tracking information on either server side or the client side. For now, the only things stored on client side is the user's nickname, user's preferred language, and the note which will automatically destroy itself after 2 days (48 hours). 
However, now I actually understood what is self-XSS. So what I should do is add a filter to the input boxes so that the users cannot enter any code that will break our javascript? I will be working on it soon, thanks to both of you.

**Comment by mj66** [comment]
@kotobukki @casprussian 
Also, FYI, our code and input DOES NOT return to our server. We only provide a static webpage with cookies enabled, and even if some code does go back to the server, i believe that GitHub's security can handle this. (Our project is hosted on GitHub).
